### PR TITLE
Refactor net/admitter/slirp/tests to use libvmi package

### PR DESCRIPTION
**Pull request by ronger4**
Signed-off-by: ronger4 <rgershbu@redhat.com>

### What this PR does
Before this PR:
Unit tests directly created and modified VM and VMI resources by manipulating their fields.
After this PR:
Unit tests have been refactored to use the libvmi package for creating and modifying VM and VMI resources, promoting readability and standardization.

Partially address: https://github.com/kubevirt/kubevirt/issues/12059

### Why we need it and why it was done in this way
Using the libvmi package simplifies the unit tests and makes them more readable.
This approach ensures a standard way of creating VM and VMI resources across tests.

The following tradeoffs were made:

- Direct manipulation of resources was replaced with utility functions, which may introduce minor overhead but enhances readability and maintainability.

The following alternatives were considered:

### Special notes for your reviewer
None

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/network/admitter/slirp_test.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ import (
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 
+	"kubevirt.io/kubevirt/pkg/libvmi"
```
Comment (EdDev) [inline pkg/network/admitter/slirp_test.go:26]: nit: Please place this in the same group as `"kubevirt.io/kubevirt/pkg/network/admitter"`.

File: pkg/network/admitter/slirp_test.go [hunk 2]
```diff
@@ -33,17 +35,16 @@ import (
...
-		}}
+		vmi := libvmi.New(
+			libvmi.WithInterface(v1.Interface{
+				Name:                   "default",
+				InterfaceBindingMethod: v1.InterfaceBindingMethod{DeprecatedSlirp: &v1.DeprecatedInterfaceSlirp{}},
+			}),
+			libvmi.WithNetwork(&v1.Network{
+				Name:          "default",
+				NetworkSource: v1.NetworkSource{Pod: &v1.PodNetwork{}},
+			}),
```
Comment (EdDev) [inline pkg/network/admitter/slirp_test.go:46]: You can use `DefaultPodNetwork()` instead, it is a bit shorter.

File: pkg/network/admitter/slirp_test.go [hunk 3]
```diff
@@ -52,17 +53,16 @@ var _ = Describe("Validate interface with SLIRP binding", func() {
...
-		}}
+		vmi := libvmi.New(
+			libvmi.WithInterface(v1.Interface{
+				Name:                   "default",
+				InterfaceBindingMethod: v1.InterfaceBindingMethod{DeprecatedSlirp: &v1.DeprecatedInterfaceSlirp{}},
+			}),
+			libvmi.WithNetwork(&v1.Network{
+				Name:          "default",
+				NetworkSource: v1.NetworkSource{Multus: &v1.MultusNetwork{NetworkName: "net"}},
+			}),
```
Comment (EdDev) [inline pkg/network/admitter/slirp_test.go:64]: I think you can use here `libvmi.WithNetwork(libvmi.MultusNetwork("default", "net"))`

File: pkg/network/admitter/slirp_test.go [hunk 4]
```diff
@@ -72,17 +72,16 @@ var _ = Describe("Validate interface with SLIRP binding", func() {
...
-		}}
+		vmi := libvmi.New(
+			libvmi.WithInterface(v1.Interface{
+				Name:                   "default",
+				InterfaceBindingMethod: v1.InterfaceBindingMethod{DeprecatedSlirp: &v1.DeprecatedInterfaceSlirp{}},
+			}),
+			libvmi.WithNetwork(&v1.Network{
+				Name:          "default",
+				NetworkSource: v1.NetworkSource{Pod: &v1.PodNetwork{}},
+			}),
```
Comment (EdDev) [inline pkg/network/admitter/slirp_test.go:83]: Same, you can use the `DefaultPodNetwork()`.

File: pkg/network/admitter/slirp_test.go [hunk 5]
```diff
@@ -92,17 +91,16 @@ var _ = Describe("Validate interface with SLIRP binding", func() {
...
```
Comment (EdDev) [inline pkg/network/admitter/slirp_test.go:102]: Same

**Review by EdDev — COMMENTED** [review]
Thank you @ronger4 .

As Slirp is now discontinued, we no longer need to invest in its leftover code.
However, as you have already invested in it, I guess we can keep the momentum and get this in.

See inline comments, nothing major.

**Comment by ronger4** [comment]
Thanks @EdDev ,
I adapted the code as you suggested.

Side note: if this is being discontinued, it means it's going to be removed? or just from a point of view that no need code is being added there anymore?

**Comment by EdDev** [comment]
> Side note: if this is being discontinued, it means it's going to be removed? or just from a point of view that no need code is being added there anymore?

Slirp is no more. The unit tests cover validations for old VM which are still running after upgrade. But even that we will drop soon.
We just do not need to invest in slirp anymore, that was my point.

**Comment by ronger4** [comment]
> > Side note: if this is being discontinued, it means it's going to be removed? or just from a point of view that no need code is being added there anymore?
> 
> Slirp is no more. The unit tests cover validations for old VM which are still running after upgrade. But even that we will drop soon. We just do not need to invest in slirp anymore, that was my point.

I was wondering why the test still there if not used anymore but you got that answered now. 
Thanks.

**Review by EdDev — COMMENTED** [review]
Thanks!

/approve

**Review by orelmisan — APPROVED** [review]
Thank you @ronger4

**Pull request merged**